tyrian/js/src/main/scala/tyrian/TyrianRoutedAppF.scala

```diff
@@ -0,0 +1,45 @@
+package tyrian
+
+import cats.effect.kernel.Async
+import cats.effect.kernel.Resource
+import org.scalajs.dom.Element
+import org.scalajs.dom.document
+import tyrian.runtime.TyrianRuntime
+
+import scala.scalajs.js.annotation._
+
+/** The TyrianApp trait can be extended to conveniently prompt you for all the methods needed for a Tyrian app, as well
+  * as providing a number of standard app launching methods.
+  */
+trait TyrianRoutedAppF[F[_]: Async, Msg, Model] extends TyrianAppF[F, Msg, Model]:
+
+  def hashResultToMessage: Navigation.Result => Msg
+  def hashChangeToMessage: Navigation.Result.HashChange => Msg
+
+  def setLocationHash(newHash: String): Cmd[F, Nothing] =
+    Navigation.setLocationHash(newHash)
```

Comment:
Is it worth enough adding this method here? I don't see any advantages to using it compared to calling `Navigation.setLocationHash` directly (and users already need to import `Navigation.Result` and others to implement hash result and change to message functions)

Reply:
Unsure, quite honestly. Mostly people will change the address by clicking a link, this is just here so you can do the same thing from a model update. I was sort of thinking "if it's hear, they don't need to know about `Navigation`, but it feels weird. Really all we're doing is setting `window.location.hash`, and I could just leave people to do that themselves. Again, deferring that decision to later.

Reply:
I have done this in the end, it's now `Routing.setLocation`. (`Navigation` has been removed.)